staging/src/k8s.io/kube-aggregator/pkg/apiserver/handler_proxy.go

```diff
@@ -178,6 +174,8 @@ func (r *proxyHandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
 	}
 
 	handler := proxy.NewUpgradeAwareHandler(location, proxyRoundTripper, true, upgrade, &responder{w: w})
+	handler.InterceptRedirects = utilfeature.DefaultFeatureGate.Enabled(genericfeatures.StreamingProxyRedirects)
+	handler.RequireSameHostRedirects = utilfeature.DefaultFeatureGate.Enabled(genericfeatures.ValidateProxyRedirects)
```

Comment:
These two flags only affect [upgraded requests](https://github.com/kubernetes/kubernetes/blob/35ae8c9fe4589bbe99e174929f78f8228c2da528/staging/src/k8s.io/apimachinery/pkg/util/proxy/upgradeaware.go#L302-L304). I added them here because the removed `maybeWrapForConnectionUpgrades` function used to set them for the SPDY transport.

They should have no effect as they are deprecated in 1.22 (https://github.com/kubernetes/enhancements/issues/1558). I keep them just for consistency.